hypervisor/src/arch/x86/emulator/instructions/cmp.rs

```diff
@@ -0,0 +1,380 @@
+//
+// Copyright © 2020 Microsoft
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#![allow(non_camel_case_types)]
+
+//
+// CMP-Compare Two Operands
+//
+
+extern crate iced_x86;
+
+use crate::arch::emulator::{EmulationError, PlatformEmulator};
+use crate::arch::x86::emulator::instructions::*;
+use crate::arch::x86::Exception;
+
+// CMP affects OF, SF, ZF, AF, PF and CF
+const CF_SHIFT: usize = 0;
+const PF_SHIFT: usize = 2;
+const AF_SHIFT: usize = 4;
+const ZF_SHIFT: usize = 6;
+const SF_SHIFT: usize = 7;
+const OF_SHIFT: usize = 11;
+
+const CF: u64 = 1 << CF_SHIFT;
+const PF: u64 = 1 << PF_SHIFT;
+const AF: u64 = 1 << AF_SHIFT;
+const ZF: u64 = 1 << ZF_SHIFT;
+const SF: u64 = 1 << SF_SHIFT;
+const OF: u64 = 1 << OF_SHIFT;
+
```

Comment:
Maybe this is something that we should move to `hypervisor/src/arch/x86/mod.rs` or `hypervisor/src/arch/x86/regs.rs` as future emulated instructions may benefit from it.

Reply:
I can do this in a follow-up PR.

Reply:
Actually I've pushed a new patch to move the bits to `regs.rs`